components/browser/search/src/main/java/mozilla/components/browser/search/provider/AssetsSearchEngineProvider.kt

```diff
@@ -46,6 +48,12 @@ class AssetsSearchEngineProvider(
         return loadSearchEnginesFromList(context, searchEngineIdentifiers.distinct())
     }
 
+    // Bug fix (hack!) ensure localization provider vars get the right locale
+    private fun enforceLocaleChange() {
+        localizationProvider.language = Locale.getDefault().language
+        localizationProvider.country = Locale.getDefault().country
```

Comment:
The localization provider can have different implementations so you do not want to always override that with the default locale here.

Reply:
I agree since localizationProvider is of "type" SearchLocalizationProvider. I hadn't thought of that